src/YamlFrontMatter.php

```diff
@@ -23,6 +23,79 @@ public static function parse(string $content): Document
         return new Document($matter, $body);
     }
 
+    /**
+     * A parser that can handle Markdown that contains Markdown.
+     *
+     * Attempts to follow the practices defined in https://jekyllrb.com/docs/front-matter/.
+     *
+     * Fixes https://github.com/spatie/yaml-front-matter/discussions/30.
+     *
+     * @param string $content
+     * @return Document
+     */
+    public static function markdownCompatibleParse(string $content): Document
+    {
+        // Turn the string into an array of lines, making the code easier to understand
+        $lines = explode("\n", $content);
+
+        // Find the line numbers of the front matter start and end blocks
+        $frontMatterControlBlockIndex = [];
+        foreach ($lines as $lineNumber => $lineContents) {
+            // If the line starts with three dashes, it's a front matter control block
+            if (substr($lineContents, 0, 3) === '---') {
+                // The line contains the front matter control block
+                if (sizeof($frontMatterControlBlockIndex) === 0) {
+                    // This is the first control block
+                    $frontMatterControlBlockIndex['start'] = $lineNumber;
+                } elseif (sizeof($frontMatterControlBlockIndex) === 1) {
```

Comment:
Could you refactor this so we don't use `elsif` and `break`?

Reply:
Done.